pkg/build/admission/overrides/api/types.go

```diff
@@ -17,4 +19,10 @@ type BuildOverridesConfig struct {
 	// If user provided a label in their Build/BuildConfig with the same name as one in this
 	// list, the user's label will be overwritten.
 	ImageLabels []buildapi.ImageLabel
+
+	// nodeSelector is a selector which must be true for the build pod to fit on a node
+	NodeSelector map[string]string
```

Comment:
it looks like these defaults are applied key-by-key? so if I have a build with `"nodeSelector": {"foo":"bar"}` and defaults of `"nodeSelector":{"baz":"qux"}`, I'll get a pod with `"nodeSelector":{"foo":"bar","baz":"qux"}`. Is that what we want for the defaulting?